File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/unpacker/ShippingQueryUnpackerParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.receiver.ShippingQueryReceiver;
import ru.ioleksiv.telegram.bot.api.model.objects.payments.ShippingQuery;
import ru.ioleksiv.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class ShippingQueryUnpackerParser extends UnpackerParser<ShippingQueryReceiver, ShippingQuery> {

    @Override
    public Class<ShippingQueryReceiver> getAnnotationClass() {
        return ShippingQueryReceiver.class;
    }

    @Override
    public UpdateUnpacker<ShippingQuery> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getShippingQuery()), ShippingQuery.class);
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/ShippingQueryUnpackerParser.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker

import io.github.oleksivio.telegram.bot.api.annotations.receiver.ShippingQueryReceiver
import io.github.oleksivio.tl.kbot.server.api.objects.payments.ShippingQuery
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker
import org.springframework.stereotype.Component

@Component
class ShippingQueryUnpackerParser : UnpackerParser<ShippingQueryReceiver, ShippingQuery>() {
    override val updateUnpacker: UpdateUnpacker<ShippingQuery>
        get() = UpdateUnpacker({ it.shippingQuery }, ShippingQuery::class)

    override val parserAnnotationClass = ShippingQueryReceiver::class

}
